### PR TITLE
Add newer OSes and Puppet 7 to test matrix, drop Puppet 5

### LINKED ISSUE #361: Test with Puppet 7 and fresher OSes
Should probably bump up the testing matrix a bit before release

### PR DESCRIPTION
**Pull request by oranenj**
Let's see what the tests say about this. Fixes #361

### CODE REVIEW

**Comment by oranenj** [comment]
I think the test setup has locale trouble. There's a line where it fails to set the locale and defaults to C, and then:
```
  Error: /Stage[main]/Postgresql::Server::Reload/Exec[postgresql_reload]: Failed to call refresh: invalid byte sequence in US-ASCII
  Error: /Stage[main]/Postgresql::Server::Reload/Exec[postgresql_reload]: invalid byte sequence in US-ASCII
```

**Comment by oranenj** [comment]
Testing by myself in a CentOS 7 VM, Puppet 7 works fine. This test failure is bogus :/

**Comment by oranenj** [comment]
Tests work if locale is forced to en_US.UTF-8

**Comment by oranenj** [comment]
@bastelfreak any idea what to do about the locale thing? Seems to require changes in modulesync... :/

**Review by bastelfreak — APPROVED** [review]

**Comment by oranenj** [comment]
Merging with the locale hack for now, submitted a PR to modulesync about it

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,5 +71,9 @@ jobs:
       - name: Run tests
         run: bundle exec rake beaker
         env:
+          # Yes, this should be in modulesync, but for now, just work around weird test failures
+          # caused by locale on CentOS 7 with Puppet 7
+          LANG: en_US
+          LC_ALL: en_US.UTF-8
```
Comment (ekohl) [inline .github/workflows/ci.yml:77]: https://github.com/voxpupuli/modulesync_config/pull/708 has more context. Linking to cross-reference.

**Pull request merged**